packages/build/src/core/constants.js

```diff
@@ -20,10 +28,11 @@ const getConstants = async function ({
   siteInfo: { id: siteId },
   token,
   mode,
-  logs,
   buildbotServerSocket,
+  logs,
 }) {
-  await validateNetlifyDir(logs, buildDir)
+  // @todo Remove once we drop support for the legact default functions directory.
+  await checkForLegacyDefaultFunctionsDir(logs, buildDir)
```

Comment:
Theoretically there's nothing stopping people from explicitcly defining `netlify-automatic-functions` in the config as being their functions directory, in which case this warning should not be printed. However, to achieve this we would need to move this logic to `@netlify/config`, because at this point we don't know whether `functions` was added by the user or by us.

@ehmicky do you have an opinion on this?

Reply:
We ran some logs few weeks ago which confirmed that virtually no one was using `netlify-automatic-functions` since we have not documented it. 

The main risk here is for users of `@netlify/plugin-nextjs` who installed it in their `package.json`. Those users will not know about `netlify-automatic-functions` since this is done under the hood. Instead, the way to fix this error would be for them to upgrade `@netlify/plugin-nextjs` (after we've released the new versions using `netlify/functions`).

I think this warning message is still safe to keep, but we might not need to move it to `@netlify/config`.

On the other hand, we might want to print a warning message in `@netlify/build` when a build is using the old version of `@netlify/plugin-nextjs`. What do you think?